crates/solver/src/settlement_rater.rs

```diff
@@ -234,12 +236,28 @@ impl SettlementRating for SettlementRater {
         let objective_value = inputs.objective_value();
         let score = match &settlement.score {
             Some(score) => match score {
-                shared::http_solver::model::Score::Score(score) => Score::Solver(*score),
-                shared::http_solver::model::Score::Discount(discount) => Score::Discounted(
-                    big_rational_to_u256(&objective_value)
-                        .unwrap_or_default()
-                        .saturating_sub(*discount),
-                ),
+                shared::http_solver::model::Score::Solver { score } => Score::Solver(*score),
+                shared::http_solver::model::Score::Discount { score_discount } => {
+                    Score::Discounted(
+                        big_rational_to_u256(&objective_value)
+                            .unwrap_or_default()
+                            .saturating_sub(*score_discount),
+                    )
+                }
+                shared::http_solver::model::Score::RiskAdjusted {
+                    success_probability,
+                    gas_amount,
+                } => {
+                    let gas_cost = gas_amount
+                        .map(|amount| u256_to_big_rational(&amount))
+                        .unwrap_or(inputs.gas_amount)
+                        * inputs.gas_price;
```

Comment:
Maybe we should somewhere more explicitly say that `gas_amount` from solver is used only for `cost_fail` calculation and not relevant for anything other. This is actually `gas_amount` in case of settlement reverts and not a `gas_amount` in case of success.

Reply:
Good catch, this is not intended! We should probably recompute the objective value in that branch!

Reply:
I've moved out the `inputs` creation outside of `match` statement since we also want to save correct `objective_value` to `RatedSettlement` (in case of RiskAdjusted score this was not true before this commit): https://github.com/cowprotocol/services/pull/1753/commits/ce08b8b0eee4c45304c60a5f733ec2f2d949642a

Reply:
Sounds good. I was worried we would then also pass in the wrong gas estimate (which is used for the eventual submission, but I saw your comment 👍 )